From ff85977303597ab9bc01f025b9ce184a3c40810f Mon Sep 17 00:00:00 2001
From: Ilies CHERGUI <ilies.chergui@gmail.com>
Date: Sun, 6 Aug 2023 18:58:54 +0100
Subject: [PATCH] Use bfd linker

Builds fails when using gold linker:

| Building ... /home/qt/boot2qt/build-jetson-agx-xavier-devkit/tmp/work/jetson_agx_xavier_devkit-poky-linux/edk2-firmware-tegra/35.2.1-r0/edk2-tegra/edk2-nvidia/Silicon/NVIDIA/Drivers/NorFlashDxe/NorFlashDxe.inf [AARCH64]
| /home/qt/boot2qt/build-jetson-agx-xavier-devkit/tmp/work/jetson_agx_xavier_devkit-poky-linux/edk2-firmware-tegra/35.2.1-r0/recipe-sysroot-native/usr/bin/aarch64-poky-linux/../../libexec/aarch64-poky-linux/gcc/aarch64-poky-linux/
12.2.0/ld: error: /home/qt/boot2qt/build-jetson-agx-xavier-devkit/tmp/work/jetson_agx_xavier_devkit-poky-linux/edk2-firmware-tegra/35.2.1-r0/edk2-tegra/edk2/BaseTools/Scripts/GccBase.lds:54:10: INFO section type is unsupported
| /home/qt/boot2qt/build-jetson-agx-xavier-devkit/tmp/work/jetson_agx_xavier_devkit-poky-linux/edk2-firmware-tegra/35.2.1-r0/recipe-sysroot-native/usr/bin/aarch64-poky-linux/../../libexec/aarch64-poky-linux/gcc/aarch64-poky-linux/
12.2.0/ld: error: /home/qt/boot2qt/build-jetson-agx-xavier-devkit/tmp/work/jetson_agx_xavier_devkit-poky-linux/edk2-firmware-tegra/35.2.1-r0/edk2-tegra/edk2/BaseTools/Scripts/GccBase.lds:66:14: INFO section type is unsupported
| /home/qt/boot2qt/build-jetson-agx-xavier-devkit/tmp/work/jetson_agx_xavier_devkit-poky-linux/edk2-firmware-tegra/35.2.1-r0/recipe-sysroot-native/usr/bin/aarch64-poky-linux/../../libexec/aarch64-poky-linux/gcc/aarch64-poky-linux/
12.2.0/ld: internal error in do_layout, at ../../gold/object.cc:1939
| collect2: error: ld returned 1 exit status

Signed-off-by: Samuli Piippo <samuli.piippo@qt.io>
Signed-off-by: Ilies CHERGUI <ilies.chergui@gmail.com>
---
 BaseTools/Conf/tools_def.template | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git edk2-tegra.a/edk2/BaseTools/Conf/tools_def.template edk2-tegra.b/edk2/BaseTools/Conf/tools_def.template
index 7ad910f1e5..a4093abc50 100755
--- edk2-tegra.a/edk2/BaseTools/Conf/tools_def.template
+++ edk2-tegra.b/edk2/BaseTools/Conf/tools_def.template
@@ -1858,7 +1858,7 @@ DEFINE GCC_AARCH64_CC_XIPFLAGS     = -mstrict-align -mgeneral-regs-only
 DEFINE GCC_DLINK_FLAGS_COMMON      = -nostdlib --pie
 DEFINE GCC_DLINK2_FLAGS_COMMON     = -Wl,--script=$(EDK_TOOLS_PATH)/Scripts/GccBase.lds
 DEFINE GCC_IA32_X64_DLINK_COMMON   = DEF(GCC_DLINK_FLAGS_COMMON) --gc-sections
-DEFINE GCC_ARM_AARCH64_DLINK_COMMON= -Wl,--emit-relocs -nostdlib -Wl,--gc-sections -u $(IMAGE_ENTRY_POINT) -Wl,-e,$(IMAGE_ENTRY_POINT),-Map,$(DEST_DIR_DEBUG)/$(BASE_NAME).map
+DEFINE GCC_ARM_AARCH64_DLINK_COMMON= -Wl,--emit-relocs -nostdlib -Wl,-fuse-ld=bfd -Wl,--gc-sections -u $(IMAGE_ENTRY_POINT) -Wl,-e,$(IMAGE_ENTRY_POINT),-Map,$(DEST_DIR_DEBUG)/$(BASE_NAME).map
 DEFINE GCC_LOONGARCH64_DLINK_COMMON= -Wl,--emit-relocs -nostdlib -Wl,--gc-sections -u $(IMAGE_ENTRY_POINT) -Wl,-e,$(IMAGE_ENTRY_POINT),-Map,$(DEST_DIR_DEBUG)/$(BASE_NAME).map
 DEFINE GCC_ARM_DLINK_FLAGS         = DEF(GCC_ARM_AARCH64_DLINK_COMMON) -z common-page-size=0x20 -Wl,--pic-veneer
 DEFINE GCC_AARCH64_DLINK_FLAGS     = DEF(GCC_ARM_AARCH64_DLINK_COMMON) -z common-page-size=0x20
-- 
2.25.1

